aliyun-ecs/src/main/java/org/jclouds/aliyun/ecs/domain/Instance.java

```diff
@@ -0,0 +1,178 @@
+            securityGroupIds == null ? ImmutableMap.<String, List<String>>of() : ImmutableMap.copyOf(securityGroupIds),
+            saleCycle, spotPriceLimit, autoReleaseTime, startTime, instanceName, description, resourceGroupId, osType,
+            osName, instanceNetworkType,
+            publicIpAddress == null ? ImmutableMap.<String, List<String>>of() : ImmutableMap.copyOf(publicIpAddress),
+            hostname, instanceType, creationTime, status,
+            tags == null ? ImmutableMap.<String, List<Tag>>of() : ImmutableMap.copyOf(tags), clusterId, recyclable,
+            regionId, gpuSpec, dedicatedHostAttribute,
+            operationLocks == null ? ImmutableMap.<String, List<String>>of() : ImmutableMap.copyOf(operationLocks),
+            InstanceChargeType, gpuAmount, expiredTime);
+   }
+
+   public abstract Map<String, List<String>> innerIpAddress();
+
+   public abstract String imageId();
+
+   public abstract String instanceTypeFamily();
+
+   public abstract String vlanId();
+
+   @Nullable
+   public abstract Map<String, List<NetworkInterface>> networkInterfaces();
+
+   public abstract String instanceId();
+
+   public abstract EipAddress eipAddress();
+
+   public abstract Integer internetMaxBandwidthIn();
+
+   public abstract String zoneId();
+
+   public abstract String internetChargeType();
+
+   public abstract String spotStrategy();
+
+   public abstract String stoppedMode();
+
+   public abstract String serialNumber();
+
+   public abstract Boolean ioOptimized();
+
+   public abstract Integer memory();
+
+   public abstract Integer cpu();
+
+   public abstract VpcAttributes vpcAttributes();
+
+   public abstract Integer internetMaxBandwidthOut();
+
+   public abstract Boolean deviceAvailable();
+
+   public abstract Map<String, List<String>> securityGroupIds();
+
+   public abstract String saleCycle();
+
+   public abstract Double spotPriceLimit();
+
+   public abstract String autoReleaseTime();
+
+   public abstract Date startTime();
+
+   public abstract String instanceName();
+
+   public abstract String description();
+
+   public abstract String resourceGroupId();
+
+   public abstract String osType();
+
+   public abstract String osName();
+
+   public abstract String instanceNetworkType();
+
+   public abstract Map<String, List<String>> publicIpAddress();
+
+   public abstract String hostname();
+
+   public abstract String instanceType();
+
+   public abstract Date creationTime();
+
+   public abstract Status status();
+
+   public abstract Map<String, List<Tag>> tags();
+
+   public abstract String clusterId();
+
+   public abstract Boolean recyclable();
+
+   public abstract String regionId();
+
+   public abstract String gpuSpec();
+
+   public abstract DedicatedHostAttribute dedicatedHostAttribute();
+
+   public abstract Map<String, List<String>> operationLocks();
+
+   public abstract String InstanceChargeType();
+
+   public abstract Integer gpuAmount();
+
+   public abstract Date expiredTime();
```

Comment:
Are all these properties always (and in every case) present? Or do we need to add some `@Nullable` annotations?